dspace-api/src/main/java/org/dspace/app/audit/MetadataEvent.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.audit;
+
+import org.dspace.content.MetadataValue;
+
+/**
+ * Represents an event related to the modification of a metadata value,
+ * storing details for audit purposes in the Solr core.
+ * This class is used to track changes such as addition, modification, or removal
+ * of metadata values, including field, value, authority, confidence, and place.
+ *
+ * @author Stefano Maffei (stefano.maffei at 4science.com)
+ */
+public class MetadataEvent {
+
+    public static final String INITIAL_ADD = "INITIAL_ADD";
+    public static final String ADD = "ADD";
+    public static final String MODIFY = "MODIFY";
+    public static final String REMOVE = "REMOVE";
+
+    private String metadataField;
+    private String value;
+    private String authority;
+    private Integer confidence;
+    private int place;
+    private String action;
+
+    public MetadataEvent() {
+    }
+
+    public MetadataEvent(MetadataValue metadataValue, String action) {
+        this.metadataField = metadataValue.getMetadataField().toString('_');
+        this.value = metadataValue.getValue();
+        this.authority = metadataValue.getAuthority();
+        this.confidence = metadataValue.getConfidence();
+        this.place = metadataValue.getPlace();
+        this.action = action;
+    }
+
+    public String getMetadataField() {
```

Comment:
Don't want to be picky about it, but when writing code, it's helpful if the IDE suggests and displays some information regarding the available class methods, even if we consider them to be self explanatory, by adding java docs to it, even if they are just gets and sets, could be helpful to understand what it's expected.